pkg/api/rest/client.go

```diff
@@ -0,0 +1,69 @@
+package rest
+
+import (
+	"net/url"
+	"strings"
+
+	"github.com/pingcap/ticdc/pkg/httputil"
+)
+
+// RESTInterface includes a set of operations to interact with cdc REST apis
+type RESTInterface interface {
+	Verb(verb string) *Request
+	Post() *Request
+	Put() *Request
+	Get() *Request
+	Delete() *Request
+}
+
+type RESTClient struct {
+	// base is the root URL for all invocations of the client.
+	base *url.URL
+
+	// versionedAPIPath is a http url prefix with api version. eg. /api/v1.
+	versionedAPIPath string
+
+	// Client is a wrapped http client.
+	Client *httputil.Client
+}
+
+// NewRESTClient creates a new RESTClient.
+func NewRESTClient(baseURL *url.URL, versionedAPIPath string, client *httputil.Client) (*RESTClient, error) {
+	base := *baseURL
+	if !strings.HasSuffix(base.Path, "/") {
+		base.Path += "/"
+	}
+	base.RawQuery = ""
+	base.Fragment = ""
+
+	return &RESTClient{
+		base:             &base,
+		versionedAPIPath: versionedAPIPath,
+		Client:           client,
+	}, nil
+}
+
+// Verb begins a request with a verb (GET, POST, PUT, DELETE).
+func (c *RESTClient) Verb(verb string) *Request {
```

Comment:
Why is this an interface method?

Reply:
Because it allows us to create a mocked RESTClient which will be convenient to do unit test. Similar as https://pkg.go.dev/k8s.io/client-go@v0.23.1/rest/fake